### PR TITLE
Test/Additional logger tests

### PR DESCRIPTION
**Pull request by MarkComerford**
#### 📲 What

A description of the change.

#### 🤔 Why

Why it's needed, background context.

#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: e2e/logger-e2e/tests/logger.spec.ts [hunk 1]
```diff
@@ -47,5 +50,15 @@ describe('logger e2e', () => {
...
             });
         });
+
+        describe('test logger', () => {
+            it('should run logger tests', async () => {
+                const project = uniq('logger');
+                await runNxCommandAsync(
+                    `generate @ensono-stacks/logger:winston ${project}`,
+                );
+                expect(await runTarget(project, targetOptions.test)).toContain('Successfully ran target test');
```
Comment (Danphillipz) [inline e2e/logger-e2e/tests/logger.spec.ts:60]: Does the logger have any other targets, linting?

Comment (MarkComerford) [reply e2e/logger-e2e/tests/logger.spec.ts:60]: It doesn't look like logger has any other commands to run, you can only do build and test.

As for other output for test failures. We should only get that text if all tests pass from what I can see. What could maybe check that it doesn't contain something, maybe something like Fail/Failed.

File: packages/logger/src/generators/winston/generator.spec.ts [hunk 2]
```diff
@@ -26,29 +46,95 @@ describe('logger generator', () => {
...
+                `custom/${appName}/tsconfig.lib.json`,
+                `custom/${appName}/tsconfig.spec.json`,
+                `custom/${appName}/package.json`,
+                `custom/${appName}/project.json`,
+                `custom/${appName}/.eslintrc.json`,
+                `custom/${appName}/jest.config.ts`,
+                `jest.config.ts`,
+                `jest.preset.js`,
+                `.prettierignore`,
+                `.prettierrc`,
```
Comment (Danphillipz) [inline packages/logger/src/generators/winston/generator.spec.ts:73]: There is some duplication here with the snapshotFiles check below.
Where we check a file in snapshotFiles (e.g project.json, tsconfig.json) we do not need to verify them within this check aswell.

The snapshotFiles function by default already calls chechFilesExistInTree on all files which you pass to it

File: packages/logger/src/generators/winston/generator.spec.ts [hunk 3]
```diff
@@ -26,29 +46,95 @@ describe('logger generator', () => {
...
-        const config = readProjectConfiguration(tree, 'custom-test-client');
+        const config = readProjectConfiguration(tree, 'custom-test-logger');
 
         expect(config).toBeDefined();
-        expect(config.tags).toEqual(['test', 'client']);
-        expect(tree.exists('test-client/src/index.ts'));
-        expect(tree.exists('test-client/src/index.test.ts'));
+        expect(config.tags).toEqual(['test', 'logger']);
+        expect(tree.exists(`${appName}/src/index.ts`));
+        expect(tree.exists(`${appName}/src/index.test.ts`));
```
Comment (Danphillipz) [inline packages/logger/src/generators/winston/generator.spec.ts:57]: Could be added to the below checkFilesExistInTree list?

**Review by Danphillipz — CHANGES_REQUESTED** [review]

File: e2e/logger-e2e/tests/logger.spec.ts (hunk 1 above)
Comment (Danphillipz) [reply e2e/logger-e2e/tests/logger.spec.ts:60]: In which case, could we also get a test added here for building?

File: packages/logger/src/generators/winston/generator.spec.ts (hunk 3 above)
Comment (MarkComerford) [reply packages/logger/src/generators/winston/generator.spec.ts:57]: Moved this into part of snapshotFiles check

File: packages/logger/src/generators/winston/generator.spec.ts (hunk 2 above)
Comment (MarkComerford) [reply packages/logger/src/generators/winston/generator.spec.ts:73]: Removed the individual checkFilesExistInTree check

File: e2e/logger-e2e/tests/logger.spec.ts (hunk 1 above)
Comment (MarkComerford) [reply e2e/logger-e2e/tests/logger.spec.ts:66]: Added additional test for build logger

**Pull request merged**